pkg/parser/parser.go

```diff
@@ -133,17 +130,26 @@ func (p *PackageParser) Parse(_ context.Context, reader io.ReadCloser) (*Package
 	return pkg, nil
 }
 
-// isWhiteSpace determines whether the passed in bytes are all unicode white
-// space.
-func isWhiteSpace(bytes []byte) bool {
-	empty := true
-	for _, b := range bytes {
-		if !unicode.IsSpace(rune(b)) {
-			empty = false
-			break
+// cleanYAML cleans up YAML by removing empty and commented out lines which
+// cause issues with decoding.
+func cleanYAML(y []byte) []byte {
+	lines := []string{}
+	for _, line := range strings.Split(string(y), "\n") {
+		trimmed := strings.TrimSpace(line)
+		if trimmed == "" || strings.HasPrefix(trimmed, "#") {
+			continue
+		}
+		lines = append(lines, line)
+	}
+	// If all left are just separators and/or terminators, then there is no
+	// content.
+	for _, c := range lines {
+		trimmed := strings.TrimSpace(c)
+		if trimmed != "---" && trimmed != "..." {
```

Comment:
nit: would it be more readable if we perform this check in the above loop and return cleaned yaml `if len(lines) > 0` ?

Reply:
The separators are not removed so there might be multiple lines with just separators but you're right we don't need the second loop. I added that emptiness check to the loop above.